.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,134 @@
+                  echo "base-branch=benchmark--$VERSION" >> $GITHUB_OUTPUT
+
+            - uses: ./.github/actions/configure_git
+              with:
+                  gpg_signing_key: ${{ secrets.GPG_SIGNING_KEY }}
+
+            - name: Create base branch
+              run: |
+                  git checkout -b ${{ steps.create-prefix.outputs.base-branch }}
+                  git push origin ${{ steps.create-prefix.outputs.base-branch }}
+
+    run-benchmarks:
+        name: ${{ matrix.benchmark_group.name }}
+        needs:
+            - get-exclude-dirs
+            - create-branch-prefix
+        strategy:
+            fail-fast: false
+            matrix:
+                benchmark_group:
+                    - { name: 'Examples', directories: './examples' }
+                    - {
+                          name: 'E2E Class',
+                          directories: './tests/end_to_end/candid_rpc/class_syntax'
+                      }
+                    - {
+                          name: 'E2E Functional',
+                          directories: './tests/end_to_end/candid_rpc/functional_syntax'
+                      }
+                    - {
+                          name: 'E2E HTTP Server',
+                          directories: './tests/end_to_end/http_server'
+                      }
+                    - {
+                          name: 'Property Class',
+                          directories: './tests/property/candid_rpc/class_api'
+                      }
+                    - {
+                          name: 'Property Functional',
+                          directories: './tests/property/candid_rpc/functional_api'
+                      }
+                    - {
+                          name: 'Property IC API',
+                          directories: './tests/property/ic_api'
+                      }
+        uses: ./.github/workflows/benchmark_parallel.yml
+        secrets:
+            GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+            GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            LASTMJS_GITHUB_TOKEN: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+        with:
+            directories: ${{ matrix.benchmark_group.directories }}
+            exclude-dirs: ${{ needs.get-exclude-dirs.outputs.exclude-dirs }}
+            branch-prefix: ${{ needs.create-branch-prefix.outputs.branch-prefix }}
+
+    finalize-benchmark:
+        needs: [run-benchmarks, create-branch-prefix]
+        uses: ./.github/workflows/squash_branches.yml
+        with:
+            branch-prefix: ${{ needs.create-branch-prefix.outputs.branch-prefix }}
+            commit-message: 'run benchmarks'
+        secrets:
+            GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+            GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            LASTMJS_GITHUB_TOKEN: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+
+    create-pr:
+        needs: [finalize-benchmark, create-branch-prefix]
```

Comment:
reduce needs to just finalize_benchmark

Reply:
nope, it is a dependencies because we need the branch prefix